browser.py

```diff
@@ -20,27 +20,27 @@
 
 import gi
 gi.require_version("Gtk", "3.0")
-gi.require_version("WebKit", "3.0")
+gi.require_version("WebKit2", "4.0")
 
 from gi.repository import Gtk
 from gi.repository import GLib
-from gi.repository import WebKit
+from gi.repository import WebKit2 as WebKit
 from gi.repository import GObject
 
 
 class Browser(Gtk.ScrolledWindow):
 
     __gsignals__ = {
         "load-finished": (
-            GObject.SIGNAL_RUN_LAST, GObject.TYPE_NONE, [GObject.TYPE_STRING]
+            GObject.SIGNAL_RUN_LAST, None, (GObject.TYPE_STRING,)
         ),
     }
 
     def __init__(self):
         Gtk.ScrolledWindow.__init__(self)
 
         self.browser = WebKit.WebView()
-        self.browser.connect("load-finished", self.__load_finished)
+        self.browser.connect("load_changed", self.__load_finished)
```

Comment:
The documented signal name is load-changed, not load_changed.

Reply:
Huge mistake, didn't see that.